core/trino-main/src/test/java/io/trino/execution/TestCreateViewTask.java

```diff
@@ -0,0 +1,329 @@
+        {
+            return materializedViewPropertyManager;
+        }
+
+        @Override
+        public Optional<CatalogName> getCatalogHandle(Session session, String catalogName)
+        {
+            if (catalogHandle.getCatalogName().equals(catalogName)) {
+                return Optional.of(catalogHandle);
+            }
+            return Optional.empty();
+        }
+
+        @Override
+        public TableSchema getTableSchema(Session session, TableHandle tableHandle)
+        {
+            return new TableSchema(tableHandle.getCatalogName(), MOCK_TABLE.getTableSchema());
+        }
+
+        @Override
+        public Optional<TableHandle> getTableHandle(Session session, QualifiedObjectName tableName)
+        {
+            if (tableName.asSchemaTableName().equals(MOCK_TABLE.getTable())) {
+                return Optional.of(
+                        new TableHandle(
+                                new CatalogName(CATALOG_NAME),
+                                new TestingTableHandle(tableName.asSchemaTableName()),
+                                TestingConnectorTransactionHandle.INSTANCE,
+                                Optional.empty()));
+            }
+            return Optional.empty();
+        }
+
+        @Override
+        public Map<String, ColumnHandle> getColumnHandles(Session session, TableHandle tableHandle)
+        {
+            return MOCK_TABLE.getColumns().stream()
+                    .collect(toImmutableMap(
+                            ColumnMetadata::getName,
+                            column -> new TestingColumnHandle(column.getName())));
+        }
+
+        @Override
+        public TableMetadata getTableMetadata(Session session, TableHandle tableHandle)
+        {
+            if ((tableHandle.getConnectorHandle() instanceof TestingTableHandle)) {
+                if (((TestingTableHandle) tableHandle.getConnectorHandle()).getTableName().equals(MOCK_TABLE.getTable())) {
+                    return new TableMetadata(new CatalogName("catalog"), MOCK_TABLE);
+                }
+            }
+
+            return super.getTableMetadata(session, tableHandle);
+        }
+
+        @Override
+        public void createTable(Session session, String catalogName, ConnectorTableMetadata tableMetadata, boolean ignoreExisting)
+        {
+            checkArgument(ignoreExisting || !tables.containsKey(tableMetadata.getTable()));
+            tables.put(tableMetadata.getTable(), tableMetadata);
+        }
+
+        @Override
+        public Optional<MaterializedViewDefinition> getMaterializedView(Session session, QualifiedObjectName viewName)
+        {
+            return Optional.empty();
+        }
+
+        @Override
+        public void createView(Session session, QualifiedObjectName viewName, ViewDefinition definition, boolean replace)
+        {
+            SchemaTableName viewSchemaTableName = viewName.asSchemaTableName();
+            if (tables.containsKey(viewSchemaTableName)) {
```

Comment:
the tests above are mostly checking this (mock) logic. If the checks are moved to CreateTableTask (see comment below) this will be unnecessary and you can extend the test class from BaseDataDefinitionTaskTest.